quickwit/quickwit-index-management/src/index.rs

```diff
@@ -481,6 +481,40 @@ impl IndexService {
         Ok(source)
     }
 
+    /// Updates a source from an index identified by its UID.
+    pub async fn update_source(
+        &mut self,
+        index_uid: IndexUid,
+        source_config: SourceConfig,
+    ) -> Result<SourceConfig, IndexServiceError> {
+        let source_id = source_config.source_id.clone();
+        check_source_connectivity(&self.storage_resolver, &source_config)
+            .await
+            .map_err(IndexServiceError::InvalidConfig)?;
+        let update_source_request =
+            UpdateSourceRequest::try_from_source_config(index_uid.clone(), &source_config)?;
+        self.metastore.update_source(update_source_request).await?;
```

Comment:
nit: The `AddSource` and `UpdateSource` source should return the added/updated source. I suspect I forgot to do so for `AddSource`. No need to tackle that in this PR, though.